docs/developer-docs/smart-contracts/compile.mdx

```diff
@@ -0,0 +1,75 @@
+---
+keywords: [beginner, tutorial, compile]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Compile
+
+<MarkdownChipRow labels={["Beginner", "Tutorial"]} />
+
+## Overview
+
+After you have written the code for your project's canisters, you need to compile the code into a WebAssembly module before it can be deployed on ICP. To build your canister's Wasm module, you can use the `dfx build` command. The `dfx build` command looks for source code to compile for each canister configured under the `canisters` section in the `dfx.json` configuration file. It can be used to build a specific canister or all canisters defined in the project's [`dfx.json`](/docs/current/developer-docs/developer-tools/cli-tools/dfx-json-reference) file.
+
+Check out the [default project template](/docs/current/developer-docs/getting-started/default-template) for more information on `dfx.json`.
+
+Before using the `dfx build` command, first you need to create the canister with the `dfx canister create <canister name>` command:
```

Comment:
It seems confusing to explain canister creation in the middle of compilation. Would it make sense to have a separate page for it. Also this one only explains canister creation locally, but the most interesting case is creating a canister on the mainnet (for which one would also need to explain cycles).

Reply:
Sure, we can move it to a separate page if you think it is worth expanding on.